tasks/market/task_find_url.go

```diff
@@ -0,0 +1,247 @@
+package market
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/curio/deps/config"
+	"github.com/filecoin-project/curio/harmony/harmonydb"
+	"github.com/filecoin-project/curio/harmony/harmonytask"
+	"github.com/filecoin-project/curio/harmony/resources"
+	"github.com/filecoin-project/curio/lib/ffi"
+	"github.com/filecoin-project/curio/lib/promise"
+)
+
+const FindURLTaskPollInterval = 3 * time.Second
+const SizeHeader = "Filecoin-Piece-RawSize"
+
+var log = logging.Logger("deal-handler")
+
+type FindURLTask struct {
+	db   *harmonydb.DB
+	sc   *ffi.SealCalls
+	api  headAPI
+	urls map[string]http.Header
+	cfg  *config.MK12Config
+
+	TF promise.Promise[harmonytask.AddTaskFunc]
+}
+
+func NewFindURLTask(db *harmonydb.DB, sc *ffi.SealCalls, cfg config.DealConfig, api headAPI) *FindURLTask {
```

Comment:
Stilll don't know what these do, but it looks a lot like tasks/market/task_find_deal.go